lib/fluent/plugin/out_forward.rb

```diff
@@ -14,42 +14,23 @@
 #    limitations under the License.
 #
 
-require 'base64'
-require 'socket'
-require 'fileutils'
-
-require 'cool.io'
-
 require 'fluent/output'
 require 'fluent/config/error'
+require 'base64'
 
-module Fluent
-  class ForwardOutputError < StandardError
-  end
-
-  class ForwardOutputResponseError < ForwardOutputError
-  end
-
-  class ForwardOutputConnectionClosedError < ForwardOutputError
-  end
+require 'fluent/compat/socket_util'
 
-  class ForwardOutputACKTimeoutError < ForwardOutputResponseError
-  end
+module Fluent::Plugin
+  class ForwardOutput < Output
+    class Error < StandardError; end
+    class ResponseError < Error; end
+    class ConnectionClosedError < Error; end
+    class ACKTimeoutError < Error; end
```

Comment:
It's intended. There are no actual benefits to make these errors with sub-super classes.
And, in fact, there are no responses under ACK timeout situation.d